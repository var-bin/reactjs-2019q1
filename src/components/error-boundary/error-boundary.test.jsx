import React from 'react';
import { shallow } from 'enzyme';

import { ErrorBoundary } from './error-boundary';

const FakeComponent = () => (
  <div>
    I am a Fake Component!
  </div>
);

describe('ErrorBoundary component:', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ErrorBoundary>
        <FakeComponent />
      </ErrorBoundary>
    );
  });

  it('Should render children if there isn\'t any error:', () => {
    expect(wrapper.is(FakeComponent)).toBeTruthy();
  });

  it('Should show error message:', () => {
    const error = new Error('hi!');

    wrapper.find(FakeComponent).simulateError(error);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
