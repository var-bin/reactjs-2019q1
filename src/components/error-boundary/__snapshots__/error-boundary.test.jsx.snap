// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component: Should show error message: 1`] = `
"<div><h2>Something went wrong.</h2><details style=\\"white-space:pre-wrap\\">Error: hi!<br/>
    in FakeComponent (created by ErrorBoundary)
    in ErrorBoundary (created by WrapperComponent)
    in WrapperComponent</details></div>"
`;
